Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -221,69 +223,81 @@ InjectorFunction dplModelAdaptor(std::vector<OutputSpec> const& filterSpecs, boo
   return [filterSpecs = std::move(filterSpecs), throwOnUnmatchedInputs, droppedDataSpecs = std::make_shared<DroppedDataSpecs>()](FairMQDevice& device, FairMQParts& parts, ChannelRetriever channelRetriever) {
     std::unordered_map<std::string, FairMQParts> outputs;
     std::vector<std::string> unmatchedDescriptions;
-    int lastSplitPartIndex = -1;
-    std::string channelNameForSplitParts;
     static int64_t dplCounter = -1;
     dplCounter++;
-    for (size_t msgidx = 0; msgidx < parts.Size() / 2; ++msgidx) {
-      const auto dh = o2::header::get<DataHeader*>(parts.At(msgidx * 2)->GetData());
+    for (size_t msgidx = 0; msgidx < parts.Size(); msgidx += 2) {
+      const auto dh = o2::header::get<DataHeader*>(parts.At(msgidx)->GetData());
```

Comment:
This line, and the same below, was the problematic line which was not adapted in 74dc3e704132fb0da641c34777337c87e5840b8a